server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -260,20 +303,29 @@ public void triggered(SchedulerEngine.Event event) {
         }
     }
 
-    // default visibility for testing purposes
-    void tryToLogHealth() {
-        if (this.currentlyRunning.compareAndExchange(false, true) == false) {
-            RunOnce release = new RunOnce(() -> currentlyRunning.set(false));
-            try {
-                ActionListener<List<HealthIndicatorResult>> listenerWithRelease = ActionListener.runAfter(resultsListener, release);
-                this.healthService.getHealth(this.client, null, false, 0, listenerWithRelease);
-            } catch (Exception e) {
-                logger.warn(() -> "The health periodic logger encountered an error.", e);
-                // In case of an exception before the listener was wired, we can release the flag here, and we feel safe
-                // that it will not release it again because this can only be run once.
-                release.run();
+    // default visibility and returns true if the semaphore was acquired, used only for testing
+    boolean tryToLogHealth() {
+        try {
+            // We only try to run this because we do not want to pile up the executions.
+            if (currentlyRunning.tryAcquire(0, TimeUnit.SECONDS)) {
+                RunOnce release = new RunOnce(currentlyRunning::release);
+                try {
+                    ActionListener<List<HealthIndicatorResult>> listenerWithRelease = ActionListener.runAfter(resultsListener, release);
+                    this.healthService.getHealth(this.client, null, false, 0, listenerWithRelease);
+                } catch (Exception e) {
+                    logger.warn(() -> "The health periodic logger encountered an error.", e);
+                    // In case of an exception before the listener was wired, we can release the flag here, and we feel safe
+                    // that it will not release it again because this can only be run once.
+                    release.run();
```

Comment:
It's very paranoid, but would you be open to switching the order of these statements? Otherwise if `logger.warn()` ever throws an exception we'll fail to release the semaphore and never be able to log again.

Reply:
Better paranoid with these things ;)